File: apps/03-food-simple/backend/python/src/services/markdown_service.py
import os
import glob
from typing import List, Dict, Any

class MarkdownService:
    def __init__(self):
        # Get the absolute path to the data directory
        self.data_dir = '/data'
    
    def read_all_markdown_files(self):
        """
        Read all markdown files from the data directory
        
        Returns:
            List[Dict[str, Any]]: An array of markdown files with their content
        """
        try:
            # Find all .md files in the data directory
            markdown_files = glob.glob(os.path.join(self.data_dir, '*.md'))
            contents = []
            for file_path in markdown_files:
                filename = os.path.basename(file_path)
                with open(file_path, 'r', encoding='utf-8') as file:
                    content = file.read()
                    contents.append({
                        'filename': filename,
                        'content': content
                    })
            
            return contents
        except Exception as e:
            print(f"Error reading markdown files: {str(e)}")
            raise 
            
    async def read_markdown_files_by_names(self, filenames: List[str]) -> List[Dict[str, Any]]:
        """
        Read markdown files by their filenames
        
        Args:
            filenames: List of filenames to read
            
        Returns:
            List[Dict[str, Any]]: An array of markdown files with their content
        """
        try:
            contents = []
            for filename in filenames:
                file_path = os.path.join(self.data_dir, filename)
                if os.path.exists(file_path):
                    with open(file_path, 'r', encoding='utf-8') as file:
                        content = file.read()
                        contents.append({
                            'filename': filename,
                            'content': content
                        })
            
            return contents
        except Exception as e:
            print(f"Error reading markdown files by names: {str(e)}")
            raise 

File: apps/03-food-simple/backend/nodejs/src/services/markdownService.js
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

class MarkdownService {
  constructor() {
    this.dataDir = '/data';
  }

  async readAllMarkdownFiles() {
    try {
      const files = await fs.readdir(this.dataDir);
      const markdownFiles = files.filter(file => file.endsWith('.md'));
      
      const contents = await Promise.all(
        markdownFiles.map(async (file) => {
          const filePath = join(this.dataDir, file);
          const content = await fs.readFile(filePath, 'utf-8');
          return {
            filename: file,
            content
          };
        })
      );

      return contents;
    } catch (error) {
      console.error('Error reading markdown files:', error);
      throw error;
    }
  }

  async readMarkdownFilesByNames(filenames) {
    try {
      const contents = await Promise.all(
        filenames.map(async (filename) => {
          const filePath = join(this.dataDir, filename);
          try {
            const content = await fs.readFile(filePath, 'utf-8');
            return {
              filename,
              content
            };
          } catch (error) {
            console.error(`Error reading file ${filename}:`, error);
            return null;
          }
        })
      );

      // Filter out any null values (files that couldn't be read)
      return contents.filter(file => file !== null);
    } catch (error) {
      console.error('Error reading markdown files by names:', error);
      throw error;
    }
  }
}

export const markdownService = new MarkdownService(); 